app/helpers/patient_helper.rb

```diff
@@ -28,6 +32,27 @@ def normalize_and_get_state_name(name)
     state_names[normalize_name(name)] || nil
   end
 
+  # This function will attempt to match the input to a language in the system
+  # PARAM: `lang` can be a three-letter iso-639-2t code, a two-letter iso-639-1 code, or the name (not case sensitive)
+  # PARAM EXAMPLES: 'eng', 'en', 'English', 'ENGLISH' <-- All will map to 'eng'
+  # RETURN VALUE: `nil` if unmatchable, else the three-letter iso code ('eng')
+  def normalize_and_get_language_name(lang)
+    return nil if lang.nil?
+
+    lang = lang.to_s.downcase.strip
+    matched_language = nil
+    matched_language = lang.to_sym if all_languages[lang.to_sym].present?
+    return matched_language unless matched_language.nil?
+
+    matched_language = all_languages.find { |_key, val| val[:display]&.casecmp(lang)&.zero? }
```

Comment:
Is this searching through the entire list of languages every time?

Reply:
Nope. it's matching the language if it can (if it's a three-letter code i.e. 'eng').

If not then yes, it does have to look through them to try and match 'english' or 'en'.

it does have short-circuit evaluation, so it's not particularly expensive imo